packages/zendesk-adapter/src/filters/audit_logs.ts

```diff
@@ -0,0 +1,315 @@
+    .forEach(prevInst => {
+      if (prevInst.annotations[CORE_ANNOTATIONS.CHANGED_BY] !== undefined) {
+        const id = prevInst.elemID.getFullName()
+        if (idByInstance[id] !== undefined) {
+          idByInstance[id].annotations[CORE_ANNOTATIONS.CHANGED_BY] = prevInst.annotations[CORE_ANNOTATIONS.CHANGED_BY]
+        }
+      }
+    })
+}
+
+const addChangedByUsingUpdatedById = (instances: InstanceElement[], idToName: Record<string, string>): void => {
+  // updated_by for translations which have updated_by_id field in the instance are not dependent on newLastAuditTime
+  instances
+    .filter(elem => TRANSLATION_TYPE_NAMES.includes(elem.elemID.typeName))
+    .forEach(elem => {
+      const id = elem.value.updated_by_id
+      if (id === undefined) {
+        log.error(`updated_by_id for the translations ${elem.elemID.name} is undefined`)
+        return
+      }
+      const name = idToName[elem.value.updated_by_id]
+      if (name === undefined) {
+        elem.annotations[CORE_ANNOTATIONS.CHANGED_BY] = DELETED_USER
+        log.error(`could not find user with id ${elem.value.updated_by_id} `)
+      } else {
+        elem.annotations[CORE_ANNOTATIONS.CHANGED_BY] = idToName[elem.value.updated_by_id]
+      }
+    })
+}
+
+const addChangedByUsingAuditLog = async ({
+  instances,
+  newLastAuditTime,
+  auditTimeInstance,
+  client,
+}:{
+  instances: InstanceElement[]
+  newLastAuditTime: string
+  auditTimeInstance: InstanceElement
+  client: ZendeskClient
+}):Promise<void> => {
+  const newLastAuditTimeMoment = moment.utc(newLastAuditTime)
+  const prevLastAuditTimeMoment = moment.utc(auditTimeInstance.value.time)
+  const updatedInstances = instances
+    // we get the changed_at from the instance itself before
+    .filter(inst => inst.annotations[CORE_ANNOTATIONS.CHANGED_AT] !== undefined)
+    .filter(inst => {
+      const instTime = moment.utc(inst.annotations[CORE_ANNOTATIONS.CHANGED_AT])
+      const isAfterPrevFetch = instTime.isAfter(prevLastAuditTimeMoment)
+      const isBeforeNewFetch = instTime.isSameOrBefore(newLastAuditTimeMoment)
+      if (!isBeforeNewFetch) {
+        // can happen for changes in ticket_fields, routing_attribute(_value), and user_fields for example
+        log.info(`There is a change that happened after the last audit time received for instance ${inst.elemID.getFullName()}`)
+      }
+      return isAfterPrevFetch && isBeforeNewFetch
+    })
+  if (_.isEmpty(updatedInstances)) {
+    return
+  }
+
+  // updated_by for everything else (some types are not supported by zendesk - listed above)
+  await awu(updatedInstances)
+    .filter(inst => !GUIDE_ELEMENTS.has(inst.elemID.typeName))
+    .forEach(async inst => {
+      const name = await getChangedByName(
+        {
+          instance: inst,
+          client,
+          start: prevLastAuditTimeMoment.format(),
+          end: newLastAuditTimeMoment.format(),
+        }
+      )
```

Comment:
nit: can be condensed to
```suggestion
      const name = await getChangedByName({
        instance: inst,
        client,
        start: prevLastAuditTimeMoment.format(),
        end: newLastAuditTimeMoment.format(),
      })
```